hbase-client/src/main/java/org/apache/hadoop/hbase/BalanceRequest.java

```diff
@@ -0,0 +1,96 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+
+/**
+ * Encapsulates options for executing an unscheduled run of the Balancer.
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Evolving
+public class BalanceRequest {
+
+  private final boolean dryRun;
+  private final boolean ignoreRIT;
+
+  private BalanceRequest(boolean dryRun, boolean ignoreRIT) {
+    this.dryRun = dryRun;
+    this.ignoreRIT = ignoreRIT;
+  }
+
+  /**
+   * Creates a BalancerRequest which runs the balancer normally.
+   * In this mode, the balancer will execute all region moves if an improved
+   * plan is found.
+   *
+   * The balancer will not run if the balance switch is disabled, the master
+   * is in maintenance mode, or there are currently regions in transition. You
+   * can force the balancer to run despite regions being in transition by
+   * using {@link #ignoreRegionsInTransition()}
+   */
+  public static BalanceRequest execute() {
+    return new BalanceRequest(false, false);
+  }
+
+
+  /**
+   * Creates a BalancerRequest which runs the balancer in dryRun mode.
+   * In this mode, the balancer will try to find a plan but WILL NOT
+   * execute any region moves or call any coprocessors.
+   *
+   * You can run in dryRun mode regardless of whether the balancer switch
+   * is enabled or disabled, but dryRun mode will not run over an existing
+   * request or chore.
+   *
+   * Dry run is useful for testing out new balance configs. See the logs
+   * on the active HMaster for the results of the dry run.
+   */
+  public static BalanceRequest dryRun() {
+    return new BalanceRequest(true, false);
+  }
+
+  /**
+   * For internal use only. Used to support deprecated interface in Admin/AsyncAdmin.
+   */
+  @InterfaceAudience.Private
+  public static BalanceRequest execute(boolean force) {
+    BalanceRequest request = BalanceRequest.execute();
+    return force ? request.ignoreRegionsInTransition() : request;
+  }
+
+  /**
+   * Updates the BalancerRequest to cause the balancer to run even if there
+   * are regions in transition.
+   *
+   * WARNING: Advanced usage only, this could cause more issues than it fixes.
+   */
+  public BalanceRequest ignoreRegionsInTransition() {
+    return new BalanceRequest(dryRun, true);
+  }
+
+  public boolean isDryRun() {
+    return dryRun;
+  }
+
+  public boolean isIgnoreRegionsInTransition() {
```

Comment:
nit: it's surprising that the name of the accessor method in a POJO does not match the underlying variable name.